dynamic_view_sizing/lib/main.dart

```diff
@@ -24,18 +24,37 @@ class _MyAppState extends State<MyApp> {
             _size += 100;
           });
         },
-        onSecondaryTap: () {
+        onDoubleTap: () {
           setState(() {
             _size -= 100;
           });
         },
-        child: Container(
-          height: _size,
-          width: _size,
-          color: Colors.yellow,
-          child: Text(
-            "I am $_size x $_size -- Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
-            style: const TextStyle(color: Colors.black),
+        child: ConstrainedBox(
+          constraints: BoxConstraints(
+            maxWidth: MediaQuery.of(context).size.width, // max-width: 100%
+            minHeight: _size,
```

Comment:
These could be applied directly as width/height of the Container below and make this simpler?

Reply:
I believe instead of doing this, we'd just want to pass appropriate constraints from the engine to the framework. Right now, I am hardcoding unconstraint constrains here: 

https://github.com/goderbauer/flutter/blob/0cf1c97f66a23a49da61d87f43cafcfbb07abae6/packages/flutter/lib/src/rendering/binding.dart#L354-L357

... because I was too lazy to wire this properly up to the engine. When we productionize this, the constraints will have to be provided by the engine; we will likely have a `FlutterView.constraints` property (which will replace `FlutterView.size`). With that property, the engine can define maxWidth, minWidth, maxHeight, and minHeight. If you want to force a width, you'd just set minWidth equal to maxWidth.

Unfortunately, I cannot try this out on my machine right now because building the web engine fails, but could you try replacing the code I linked to above with

```dart
constraints: const BoxConstraints(minWidth: view.size.width, maxWidth: view.size.height),
```

then undo the changes here and try running again. I suspect, it will give you the result you want?

Reply:
And for my own understanding: You want the fixed width because it _feels_ better that way on the web? Or are there any technical limitations that would require us on the web to always have a fixed width?

Reply:
I verified that the MediaQuery business isn't necessary if the appropriate constraints with a fixed-width are used, as described above.

Reply:
Yes, you're right. In the current implementation, a normal element is able to expand to 100% of the width. Not so in the `dynamic-view-sizing-prototype` branch, though.